OneOneOne/OneOneOne/message_handler.py

```diff
@@ -0,0 +1,190 @@
+from lxml import etree, objectify
+import xml.etree.ElementTree as ET
+
+from builder.pystache_message_builder import PystacheMessageBuilder
+from utilities.file_utilities import FileUtilities
+from definitions import XML_PATH, TEMPLATE_PATH
+import logging
+
+basic_success_response = FileUtilities.get_file_string(XML_PATH / 'basic_success_response.xml')
+
+base_fault_response = FileUtilities.get_file_string(XML_PATH / 'basic_fault_response.xml')
+
+
+def build_error_message(error):
+    builder = PystacheMessageBuilder(str(TEMPLATE_PATH), 'base_error_template')
+    return builder.build_message({"errorMessage": error})
+
+
+class MessageHandler:
+    namespaces = {
+        'soap': 'http://schemas.xmlsoap.org/soap/envelope/',
+        'a': 'http://www.etis.fskab.se/v1.0/ETISws',
+        'wsa': 'http://www.w3.org/2005/08/addressing',
+        'itk': 'urn:nhs-itk:ns:201005'
+    }
+
+    def __init__(self, message_string):
+        logging.basicConfig(level=logging.DEBUG)
+        self.message = message_string
+        self.message_tree = ET.fromstring(self.message)
+
+        self.check_list = [
+            self.check_action_types,
+            self.check_manifest_and_payload_count,
+            self.check_manifest_count_against_actual,
+            self.check_payload_count_against_actual,
+            self.check_payload_id_matches_manifest_id
+        ]
+
+    def evaluate_message(self):
+        """
+        This iterates over the message check methods searching for errors in the message, if no errors are found
+        a success response is returned
+
+        :return:
+        """
+        for check in self.check_list:
+            status, response = check()
+            if status != 200:
+                return status, response
+
+        return 200, basic_success_response
```

Comment:
In general I am a little confused as to why these returns are returning HttpStatus codes. Can this return something like a flag called `valid` since it looks like an evaluation is done. I don't think the Http needs to leak into here.

Reply:
Agreed, this class doesn't need to know it's being invoked via an HTTP request.

Reply:
I've updated these methods to use a pass/fail flag and kept instances of the error code in the message handler and the actual message content evaluation methods have been moved to their own classes as per Gareth's suggestion.

Reply:
Also would it be worth altering the response string to be just the error message and not the full message, this would simplify the functionality to just producing an error string or is it fine as it is?

Reply:
I think you still need to extract the 200/500 code from `MessageHandler.evaluate_message()`. So it would also return the pass/fail flag and it would be up to `MessageReceiver` to decide which HTTP status code to assign to a pass or a fail.

Also, I think it makes sense to keep the message building within the MessageHandler/Check classes. It's not something `MessageReceiver` needs to be responsible for.

Reply:
Ok I've updated the handler to have no reference to HTTP codes, it just returns an error flag now and the `MessageReceiver` now makes this decision as you suggested. I've extracted the message building from the individual check classes to the `MessageHandler` class as well, both the `MessageHandler` and `Check` classes have separate tests as well now.

Reply:
👍 This looks lot better to me now